### ISSUE TITLE
LHN- All chat icons turn green for a moment when user logs into High traffic account

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#8212]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Pre-Condition: Have a couple/several chats already open with other accounts
1. Access http://staging.new.expensify.com on mweb iOS
2. Log in to a high traffic account

## Expected Result:
When the user is logged in, he/she expects to see the correct color user icons next to each chat window

## Actual Result:
When the user logs in, all chats are shown with a green icon to the left of the name and after a couple of second the chat images switch to the actual colors (Yellow, blue, green, pink, etc.)

## Workaround:
Unknown 

## Platform:

Where is this issue occurring?

- Mobile Web

**Version Number:** 1.1.44.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** applausetester+ebezerra@applause.expensifail.com

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/158892837-6e0ed17a-f545-4cf3-83cb-80f6511b2a9a.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @aldo-expensify (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by kbecciv** [comment]
@aldo-expensify Yes, account with more than 10K users.

**Comment by aldo-expensify** [comment]
I this only happening in mweb IOS? can you confirm if it happens in web (desktop) or native mobile?

**Comment by kbecciv** [comment]
@aldo-expensify Able to reproduce in Andriod app
![Screenshot_20220317-212324_New Expensify](https://user-images.githubusercontent.com/93399543/159051754-66260119-fb9e-4db0-bc5b-8e881f9c26cb.jpg)

**Comment by aldo-expensify** [comment]
This is related to this code: https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/actions/PersonalDetails.js#L166-L211

The API command `PersonalDetails_GetForEmails` load some info like `avatar`, `avatarThumbnail`, `firstName`, etc. for a list of emails. I'll check if this can be fixed by improving the performance on the API side.

**Comment by aldo-expensify** [comment]
I tested locally if the `PersonalDetails_GetForEmails` was slow to responde from the API, and it is fast (tested requesting data for 500 accounts), also, from looking at the code, I think it should be fast.

I think the problem may be that we are waiting for other requests to finish first before sending `PersonalDetails_GetForEmails` and this may be improved in the front.

<img width="800" alt="image" src="https://user-images.githubusercontent.com/87341702/159760654-30225805-312f-4ccf-9b9e-42d05dcb6434.png">

**Comment by JmillsExpensify** [comment]
Upwork job posted here: https://www.upwork.com/jobs/~011b8d95c6d7c31fe1

**Comment by JmillsExpensify** [comment]
Still waiting on contributors.

**Comment by JmillsExpensify** [comment]
Still waiting on contributors. I'm going to go ahead and update the price to $1000 since I missed updating to $500 last week.

**Comment by JmillsExpensify** [comment]
Still waiting for proposal so I'm increasing the price to $2,000.

**Comment by chiragsalian** [comment]
Still waiting on proposals.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by JmillsExpensify** [comment]
I'll keep this open though I'm going to put it on hold. If we have another week where this isn't reproducible, I'll close.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by JmillsExpensify** [comment]
Going to close this one since two weeks with no regression. Let's re-open if it crops up again.
